c7n/resources/keyspaces.py

```diff
@@ -0,0 +1,302 @@
+from c7n.actions import BaseAction
+from c7n.manager import resources
+from c7n.query import ChildResourceManager, DescribeSource, QueryResourceManager, TypeInfo
+from c7n.resources.aws import shape_schema
+from c7n.tags import RemoveTag, Tag, TagActionFilter, TagDelayedAction
+from c7n.utils import get_retry, local_session, type_schema
+
+
+SYSTEM_KEYSPACES = [
+    "system",
+    "system_schema",
+    "system_schema_mcs",
+    "system_multiregion_info",
+]
+
+
+class DescribeKeyspaces(DescribeSource):
+
+    def get_permissions(self):
+        perms = super().get_permissions()
+        perms.remove('cassandra:GetKeyspace')
+        return perms
+
+    def get_resources(self, resource_ids, cache=True):
+        return [
+            r for r in super(DescribeKeyspaces, self).get_resources(resource_ids, cache)
+            if r['keyspaceName'] not in SYSTEM_KEYSPACES
+        ]
+
+    def resources(self, query):
+        return [r for r in super(DescribeKeyspaces, self).resources(query)
```

Comment:
nit: the 0-argument form of `super()` works and is already used in `get_permissions()` above. might be more consistent to use that here too, but... definitely a nit :sweat_smile: